### ISSUE TITLE
JSX nodes/attributes only?

### ISSUE THREAD

**Issue by bignimbus** [edvardchen/eslint-plugin-i18next#17]
Thank you for the wonderful plugin!

I was wondering whether there's any support for an option or rule that targets only JSX nodes/attributes and ignores all other string literals:

```jsx
import { MyOtherComponent } from "./my-other-component";

const MyComponent = () => {
  const foo = "Foo"; // no violation
  return (
    <MyOtherComponent bar="Bar">{/* violation - string literal in a JSX prop */}
      Baz {/* violation - undecorated JSX text node */}
    </MyOtherComponent>
  );
};
```

**Comment by edvardchen** [comment]
I don't think it is a good idea. 

We use a lot of variables not just plain text in `JSX` to display information. And we use many reusable components then we would pass props to them. 

Just validating the text node in `JSX` seems to be lazy and unreliable.

**Comment by bignimbus** [comment]
I agree with the opinions that are expressed in this plugin regarding strings as a general best practice. When running this against a large scale production react app, though, a prohibitive number of reported violations have nothing to do with user-facing content that can be internationalized. Anyhow, I know that’s not your problem, so thanks again for making this awesome plugin! 🙏🏻

**Comment by edvardchen** [comment]
It's terrible to see so many warnings when people import this plugin into their project first time, like the first time we use jslint or eslint. And that would scare people away.

I think your suggesting option is reasonable and it will make more developers to use the plugin. 

**Let us agree that the new option will tell the plugin to validate the text in markup (`JSX` or `VUE template`) only**

So what is the new option name? `textOnly` or `textInMarkUpOnly` or others?

@bignimbus

**Comment by bignimbus** [comment]
🎉 Either of those might work - I think `textInMarkUpOnly` is more descriptive.  I might also suggest `domNodesAndAttributes` or something like that.

**Comment by gigashiro** [comment]
Hi guys, this plugin is really amazing!

I think that is more reasonable if you could pass the file extensions list in options.

I am facing an issue in my `utils` file as seen below:
```
  return new Intl.NumberFormat(language, {
    style: 'currency',
    currency,
  }).format(value || 0)

```
Here the linter points to 'currency' as error

**Comment by edvardchen** [comment]
@gigashiro I think the option that we had discussed above can help your case. 

I'm kinda busy these days. I will try to finish the proposal this weekend.

**Comment by edvardchen** [comment]
@bignimbus Let me check again. Should we just validate the text node only, or including the JSX attributes? 

I guess the first one would be more welcome. And I don't want to add two options to distinguish this two cases.

**Comment by bignimbus** [comment]
I think both attributes and text nodes are valuable. Lots of components accept attributes for label text, for instance.

**Comment by bignimbus** [comment]
To clarify: I think linting JSX attributes is just as useful as text nodes.  We can still declare allowed attributes in the configuration, yes?

**Comment by edvardchen** [comment]
@bignimbus What about the expression?

```jsx
<ul>
{[].map(item=>(
<li>{item}</li>
))}
</li>
```

**Comment by bignimbus** [comment]
@edvardchen since `item` would not be a string literal or plain text node in that example, I’d say that shouldn’t be a violation

**Comment by edvardchen** [comment]
@bignimbus But `"title"` did.

**Comment by edvardchen** [comment]
@bignimbus I mean, should we still validate the possible express in JSX, or just validate the text node and elm attritubes?

**Comment by bignimbus** [comment]
IMO, text node and elm attributes.  Ex:

```jsx
{/* violations */}
<ul>
  {[{ title: "foo" }].map(item=>(
    <li bar="baz">My Title</li>
  ))}
</li>
```

```jsx
{/* no violations */}
<ul>
  {[{ title: "foo" }].map(item=>(
    <li foo={item.title}>{item.title}</li>
  ))}
</li>
```

**Comment by bignimbus** [comment]
The following, however, I'm not very opinionated about:

```jsx
<MyComponent foo={"foo"}>{"bar"}</MyComponent>
```

Many linter rules separately police interpolated string literals, so I'm not too worried about a loophole wherein a text node or attribute is an interpolated expression that evaluates to a plain string literal.

**Comment by edvardchen** [comment]
Check this branch's [README](https://github.com/edvardchen/eslint-plugin-i18next/tree/markup-only#markuponly)

See the `markupOnly` option

@bignimbus

**Comment by bignimbus** [comment]
@edvardchen looks good to me - I can try checking out this branch and running against a codebase, if that would help

**Comment by edvardchen** [comment]
release as 3.8.0
